src/BlockTypes/ProductQuery.php

```diff
@@ -193,6 +224,29 @@ private function get_on_sale_products_query() {
 		);
 	}
 
+	/**
+	 * Return query params to support custom sort values
+	 *
+	 * @param string $orderby  Sort order option.
+	 *
+	 * @return array
+	 */
+	private function get_custom_orderby_query( $orderby ) {
+		if ( ! in_array( $orderby, $this->custom_order_opts, true ) ) {
+			return array( 'orderby' => $orderby );
```

Comment:
Should we aim for consistency here and ensure that the array returned always have the same keys?

Currently, this function can return either `[ 'orderby' => $orderby ]` or `[ 'meta_key' => {value}, 'orderby'  => {value} ]` , depending on the conditionals.

Reply:
Well, the point here is that we have to build a custom query when we have `orderby` props which are not supported by WordPress API natively. Because of this, we return the normal query shape if it's not a custom WooCommerce-only option and we modify it in case it is. That's why we need two different shapes.

Reply:
👍 ok, thanks for clarifying.